pkg/component/worker/kubelet.go

```diff
@@ -170,15 +192,25 @@ func (k *Kubelet) Start(ctx context.Context) error {
 		args["--node-labels"] = strings.Join(k.Labels, ",")
 	}
 
-	if runtime.GOOS == "windows" {
-		node, err := node.GetNodename("")
+	nodename, err := node.GetNodename("")
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+
+	if k.DualStackEnabled {
+		ipv4, ipv6, err := getIPs(ctx, nodename)
 		if err != nil {
-			return fmt.Errorf("can't get hostname: %w", err)
+			logrus.WithError(err).Infof("failed to get ip address of node %s", nodename)
```

Comment:
```suggestion
			logrus.WithError(err).Errorf("failed to lookup %q", nodename)
```

Reply:
Kubelet uses `can't get ip address of node %s. error: %v`.
https://github.com/kubernetes/kubernetes/blob/0cc57258c3f8545c8250f0e7a1307fd01b0d283d/pkg/kubelet/nodestatus/setters.go#L231

And I think calico will work just fine in this situation, which is why I made it info only.